src/main/java/edu/kit/provideq/toolbox/Solution.java

```diff
@@ -0,0 +1,96 @@
+package edu.kit.provideq.toolbox;
+
+import java.util.Objects;
+
+/**
+ * A solution holds all information concerning a specific
+ * {@link edu.kit.provideq.toolbox.meta.Problem} solving process triggered by a
+ * {@link edu.kit.provideq.toolbox.meta.ProblemSolver}. This includes meta data, debug data,
+ * the current status of the process, as well as the eventually generated solution data
+ * @param <S> the type of the generated solution data
+ */
+public class Solution<S> {
+  private final long ID;
+  private SolutionStatus status = SolutionStatus.COMPUTING;
+  private String metaData = "";
+  private S solutionData;
+  private String debugData;
+
+  private boolean completed = false;
```

Comment:
Minor nitpick: This bool could be replaced by a function isCompleted() in SolutionStatus, because it is directly tied to the status value.